openverse_catalog/dags/maintenance/add_license_url.py

```diff
@@ -0,0 +1,162 @@
+from common.loader.sql import RETURN_ROW_COUNT
+from common.slack import send_message
+from psycopg2._json import Json
+
+
+DAG_ID = "add_license_url"
+UPDATE_LICENSE_URL = "update_license_url"
+FINAL_REPORT = "final_report"
+
+ALERT_EMAIL_ADDRESSES = ""
+
+logger = logging.getLogger(__name__)
+
+base_url = "https://creativecommons.org/"
+
+
+def get_null_counts(postgres_conn_id: str) -> int:
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    null_meta_data_count = postgres.get_first(
+        dedent("SELECT COUNT(*) from image WHERE meta_data IS NULL;")
+    )[0]
+    return null_meta_data_count
+
+
+def get_statistics(
+    postgres_conn_id: str,
+) -> Literal[UPDATE_LICENSE_URL, FINAL_REPORT]:
+    null_meta_data_count = get_null_counts(postgres_conn_id)
+
+    next_task = UPDATE_LICENSE_URL if null_meta_data_count > 0 else FINAL_REPORT
+    logger.info(
+        f"There are {null_meta_data_count} records with NULL in meta_data. \n"
+        f"Next step: {next_task}"
+    )
+
+    return next_task
+
+
+def update_license_url(postgres_conn_id: str) -> dict[str, int]:
+    """Add license_url to meta_data batching all records with the same license.
+    :param postgres_conn_id: Postgres connection id
+    """
+
+    logger.info("Getting image records without license_url.")
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    license_map = get_reverse_license_path_map()
+
+    total_count = 0
+    total_counts = {}
+    for license_items, path in license_map.items():
+        license_name, license_version = license_items
+        logger.info(f"Processing {license_name} {license_version}, {license_items}.")
+        license_url = f"{base_url}{path}/"
+
+        select_query = dedent(
+            f"""
+            SELECT identifier FROM image
+            WHERE (
+            meta_data is NULL AND license = '{license_name}'
+            AND license_version = '{license_version}')
+            """
+        )
+        result = postgres.get_records(select_query)
+
+        if not result:
+            logger.info(f"No records to update with {license_url}.")
+            continue
+        logger.info(f"{len(result)} records to update with {license_url}.")
+        license_url_col = {"license_url": license_url}
+        update_license_url_query = dedent(
+            f"""
+            UPDATE image
+            SET meta_data = {Json(license_url_col)}
+            WHERE identifier IN ({','.join([f"'{r[0]}'" for r in result])});
+            """
+        )
+
+        updated_count = postgres.run(
+            update_license_url_query, autocommit=True, handler=RETURN_ROW_COUNT
+        )
+        logger.info(f"{updated_count} records updated with {license_url}.")
+        total_counts[license_url] = updated_count
+        total_count += updated_count
+
+    logger.info(
+        f"{total_count} image records with missing license_url updated.\n"
+        f"Total updated counts: {total_counts}"
```

Comment:
Not necessary, but this could be easier to read if the `total_counts` are separated by line breaks.